ang/crmMosaico/TemplateItem.js

```diff
@@ -12,13 +13,26 @@
       },
       link: function($scope, $el, $attr) {
         var ts = $scope.ts = CRM.ts('mosaico');
-        $scope.$watch('crmMosaicoTemplateItem', function(newValue){
+
+        $scope.$watch('crmMosaicoTemplateItem', function(newValue) {
           $scope.myOptions = newValue;
+
+          // Template default action label based on current state
+          if (newValue['state']) {
+            var clickActionLabels = {
```

Comment:
You are re-defining `clickActionLabels` every time `crmMosaicoTemplateItem` changes, which i don't think it's what you want. Please define the object at the top of `link`

```js
link: function($scope, $el, $attr) {
  var clickActionLabels = {
    // ...
  }
```